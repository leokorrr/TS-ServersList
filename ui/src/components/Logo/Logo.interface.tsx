export default interface Logo {
    title: string;
}