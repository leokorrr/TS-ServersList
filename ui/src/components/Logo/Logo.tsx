import React from 'react'
import LogoInterface from './Logo.interface'

const Logo: React.FC<LogoInterface> = ({title}) => {
    return (
        <div className="logo">
            <div className="logo__circle"></div>
            <div className="logo__name">{title}</div>
            <div className="logo__line"></div>
        </div>
    )
}

export default Logo