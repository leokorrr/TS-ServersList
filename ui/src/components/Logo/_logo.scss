.logo {
    display: flex;
    align-items: center;
    margin-left: 2.7rem;
    font-family: 'Montserrat', sans-serif;

    &__name {
        color: $color-white;
        font-size: 1.4rem;
        font-weight: bold;
    }

    &__circle {
        width: 1.3rem;
        height: 1.3rem;
        border: 0.2rem solid $color-logo-circle;
        border-radius: 2rem;
        margin-right: 1.2rem;
    }

    &__line {
        height: 3.5rem;
        width: 0.2rem;
        margin-left: 3.1rem;
        background-color: $color-grey;
    }
}