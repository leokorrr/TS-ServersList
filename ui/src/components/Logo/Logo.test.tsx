import React from 'react'
import Logo from './Logo'
import { shallow, mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

const logoText = 'Recruitment task'
configure({ adapter: new Adapter() })
it("Logo rendered without crashing", () => {
    shallow(<Logo title={logoText}/>);
})
it("should render initial layout", () => {
    const component = shallow(<Logo title={logoText} />);
    expect(component.getElements()).toMatchSnapshot();
})
describe("", () => {
    it("accepts Navbar props", () => {
        const wrapper = mount(<Logo title={logoText}/>)
        expect(wrapper.props().title).toEqual(logoText)
    })
    it("contains logo text", () => {
        const wrapper = mount(<Logo title={logoText}/>)
        const value = wrapper.find("div.logo__name").text()
        expect(value).toEqual("Recruitment task")
    })
})