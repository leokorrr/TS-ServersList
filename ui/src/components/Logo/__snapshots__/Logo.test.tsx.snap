// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render initial layout 1`] = `
Array [
  <div
    className="logo"
  >
    <div
      className="logo__circle"
    />
    <div
      className="logo__name"
    >
      Recruitment task
    </div>
    <div
      className="logo__line"
    />
  </div>,
]
`;
